maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluatorV4.java

```diff
@@ -164,7 +164,7 @@ public Object evaluate(String expr, Class<?> type) throws ExpressionEvaluationEx
 
             // Was not an expression
             if (expression.contains("$$")) {
```

Comment:
I think we should replace those 5 lines with the simple call:
```
return expression.replace("$$", "$");
```
as the `String.replace()` method should exit fast if the searched string can not be found.